web/source/osk/visualKeyboard.ts

```diff
@@ -1491,19 +1523,48 @@ namespace com.keyman.osk {
       kbd.style.border = '1px solid #ccc';
 
       // Once the element is inserted into the DOM, refresh the layout so that proper text scaling may apply.
-      const refreshInterval = window.setInterval(function () {
-        let computedStyle = getComputedStyle(kbd);
-        if (computedStyle.fontSize) {
-          if (kbd.style.fontSize) {
-            // Preserve the new setting (provided by CSS)
-            kbdObj.fontSize = new ParsedLengthStyle(kbd.style.fontSize);
+      const detectAndHandleInsertion = () => {
+        if(document.contains(kbd)) {
+          // Yay, insertion!
+
+          try {
+            // Are there font-size attributes we may safely adjust?  If so, do that!
+            if(getComputedStyle(kbd) && kbd.style.fontSize) {
+              kbdObj.fontSize = new ParsedLengthStyle(kbd.style.fontSize);
+            }
+
+            // Make sure that the stylesheet is attached, now that the keyboard-doc's been inserted.
+            // The stylesheet is currently built + constructed in the same code that attaches it to
+            // the page.
+            kbdObj.appendStyleSheet();
+
+            // Grab a reference to the stylesheet.
+            const stylesheet = kbdObj.styleSheet;
+            const stylesheetHead = stylesheet.parentElement;
```

Comment:
```suggestion
            const stylesheetParentElement = stylesheet.parentElement;
```

I got confused by `head` :grin:

Reply:
Well... in the current implementation, it quite literally is `<head>` for the active document.  Can still change the name over, though.

Reply:
Sure. But that's secret implementation knowledge :grin: